lib/zendesk_api/resources.rb

```diff
@@ -464,6 +464,60 @@ class Ticket < Resource
     extend UpdateMany
     extend DestroyMany
 
+    # Proxy to trap array operator usage on custom_field_symbol
+    class CustomFieldSymbolProxy
+      def initialize(ticket, _arr)
+        @ticket = ticket
+        @field_array = @ticket.custom_fields || []
+      end
+
+      def [](key)
+        raise "Cannot find custom field #{key}, configuration ticket_fields_metadata is OFF" unless
+          @ticket.instance_variable_get("@client").ticket_fields_metadata
+        # Trap read access
+        fld = @ticket.instance_variable_get("@client").ticket_fields_metadata.find { |val| val[:title] == key }
+        raise "Cannot find custom field #{key}" unless fld
+        cf = @ticket.custom_fields.find { |h| h[:id] == fld[:id] }
+        cf ? cf[:value] : nil
+      end
+
+      def []=(key, value)
+        raise "Cannot find custom field #{key}, configuration ticket_fields_metadata is OFF" unless
+          @ticket.instance_variable_get("@client").ticket_fields_metadata
+        # Trap write access
+        fld = @ticket.instance_variable_get("@client").ticket_fields_metadata.find { |val| val[:title] == key }
+        raise "Cannot find custom field #{key}" unless fld
+        cf = @ticket.custom_fields.find { |h| h[:id] == fld[:id] } if @ticket.custom_fields
+        if cf
+          cf[:value] = value
+        else
+          @ticket.custom_fields << { id: fld[:id], value: value }
+        end
+      end
+
+      def to_a
+        @field_array
+      end
+
+      # Delegate other hash methods as needed
+      def method_missing(method, ...)
+        @field_array.send(method, ...)
+      end
+
+      def respond_to_missing?(method, include_private = false)
+        @field_array.respond_to?(method, include_private)
+      end
+    end
+
+    def custom_field_symbol
+      @custom_field_symbol ||= CustomFieldSymbolProxy.new(self, @custom_field_symbol)
+    end
+
+    def custom_field_symbol=(val)
+      @custom_field_symbol = val
+      @custom_field_symbol_proxy = CustomFieldSymbolProxy.new(self, @custom_field_symbol)
+    end
```

Comment:
`custom_field_symbol=` overwrites `@custom_field_symbol` with the raw array and caches a proxy in `@custom_field_symbol_proxy`, but the getter still memoizes on `@custom_field_symbol`. After the first assignment (or any mass-assignment that sets this ivar) subsequent reads return the plain array and `ticket.custom_field_symbol['foo']` will raise a `TypeError` when trying to index an array with a string.